lib/Varien/Autoload.php

```diff
@@ -91,7 +91,7 @@ public function autoload($class)
         }
         $classFile.= '.php';
         //echo $classFile;die();
-        return include $classFile;
```

Comment:
we dont have a `if file_exists()` check here yet, thats something we should keep in mind as a future change

Reply:
I think suppressing the `include` warning is functionally equivalent to using a conditional on `file_exists`, right?

Reply:
except you set scream.enabled=true, which makes all the @ ignored

Reply:
Then it seems like a good solution, the user that wants to see all warnings can see them and everyone else will not. Also I just remembered that `file_exists` does not search the include path like `include` so they are not functionally equivalent anyway. You could use `stream_resolve_include_path` but I still think `@include` is fine.

Reply:
interesting, I never faced a problem because of the include path not respected in this check.
The problem with this is, thats not a warning which just lands in the log, but it breaks any application which somewhere relies on a `class_exist()` which should return `false`, because if you set `scream.enabled=true`, the application will not work anymore.

And this added error supressor now makes sense, some of the new added code relies on class_exists, so it would break if the error is triggered.

anyway, we should move this to a separate discussion and decide later. The PR is good so far.

Reply:
Hi, we are facing a lot of issues in our shops with this error suppressing introduced. 
I don't think it is a good idea to suppress all kind of errors which could be triggered when including a file. Fatal errors like ` Declaration of xx must be compatible with yy` and `Cannot redeclare` will also be silently suppressed now and the application will stop excecuting resulting in a blank page (or partial HTML). The fatal error is nowhere to be found because it is suppressed. This makes debugging very time consuming and we will not get noticed if a customer experiences this kind of error. There are legitimate cases to use the @ sign, but autoloading is definitely not a place where you want to use it imo. 
`scream.enabled=true` on production environments is definitely a no go, so there is no way for us to log this errors on production.
I am not really sure which issue you are trying to resolve with the error suppressing. Is the `include` behaviour different in PHP7.2?
Please give me your thoughts about this. There must be another solution as all other autoloaders (Composer, Symfony etc.) will also not suppress include errors, and they have no problems with `class_exists()` calls.

Reply:
Why introduce error suppressing for one of the most critical components in an application. If something is wrong with code being included or some file is missing it should be abundantly clear as in almost all cases it will result in the application not being able to function correctly.

Reply:
These are great points. @cieslix do you know the reason this change was introduced specifically?